test/compress/issue-203.js

```diff
@@ -26,9 +26,7 @@ compress_new_function_with_destruct: {
     }
     expect: {
         Function("a", "[b]", "return a");
-        Function("a", "{bb}", "return a");
-        Function("[[a]]", "[{bb}]", 'return a');
+        Function("a", "{bb:b}", "return a");
+        Function("[[a]]", "[{bb:b}]", 'return a');
```

Comment:
The optimisation I mentioned in https://github.com/mishoo/UglifyJS2/pull/2029#discussion_r119141537 will indiscriminately mangle the inner function code, and with pretty much default options as well to boot.